### PR TITLE
build(deps): update opentelemetry

### PR DESCRIPTION
**Pull request by fabriziosestito**
## Description

Update opentelemetry dependencies and adapt existing code to the new API.

Dependencies updated:
[TODO] insert list here

- [x] check serializer feature (which was removed in the new version) (removed here: https://github.com/open-telemetry/opentelemetry-rust/pull/738)

Resolves: #493 

## Test

- [x] tested locally with containers
- [x] test on kubernetes

## Additional Information

-

### CODE REVIEW

**Review by flavio — APPROVED** [review]
Nice, thanks for the hard work. Reverse engineering all the breaking changes done is not easy...

File: DEVELOPMENT.md [hunk 1]
```diff
@@ -37,22 +37,21 @@ Start the OpenTelemetry collector:
...
+  otel/opentelemetry-collector:0.84.0 \
+  --config /etc/otel/config.yaml
 ```
 
 Start prometheus, so it can start scraping metrics. By adding the `host.docker.internal`, the
 prometheus container will be able to reach the OpenTelemetry collector exposed port in the host, and
 scrape that endpoint. Check the `prometheus.yml` configuration for more details.
 
-```console
+````console
```
Comment (jvanz) [inline DEVELOPMENT.md:49]:
```suggestion
```console
```

File: src/metrics/policy_evaluations_latency.rs [hunk 2]
```diff
@@ -1,13 +1,13 @@
...
-use opentelemetry::{metrics::ValueRecorder, KeyValue};
+use opentelemetry::{metrics::Histogram, KeyValue};
 use std::convert::TryFrom;
 use std::time::Duration;
 
 lazy_static! {
-    static ref POLICY_EVALUATION_LATENCY: ValueRecorder<u64> =
+    static ref POLICY_EVALUATION_LATENCY: Histogram<u64> =
         opentelemetry::global::meter(super::METER_NAME)
-            .u64_value_recorder("kubewarden_policy_evaluation_latency_milliseconds")
```
Comment (jvanz) [inline src/metrics/policy_evaluations_latency.rs:10]: Interesting. The new names make more sense in the context of Open telemetry.

**Review by jvanz — APPROVED** [review]

**Review by flavio — APPROVED** [review]
Thanks for this update. LGTM

**Pull request merged**